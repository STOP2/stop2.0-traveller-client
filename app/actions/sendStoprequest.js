import config from '../config/config'
import strings from '../resources/translations'

export const SEND_STOPREQUEST = 'SEND_STOPREQUEST'
export const RECEIVE_CONFIRM = 'RECEIVE_CONFIRM'
export const REQUEST_ERROR = 'REQUEST_ERROR'

const API_ENDPOINT = '/stoprequests'

export let requestStoprequest = function(tripId, stopId, requestType)
{
    return {
        type: SEND_STOPREQUEST,
        sentStoprequest: false,
        tripId: tripId,
        stopId: stopId,
        requestType: requestType
    }
}

export let receiveConfirm = function()
{
    return {
        type: RECEIVE_CONFIRM,
        sentStoprequest: true
    }
}

export let requestError = function()
{
    alert(strings.stopRequestError)

    return {
        type: REQUEST_ERROR,
        error: true
    }
}

export let sendStoprequest = function(tripId, stopId, requestType)
{
    return dispatch =>
    {
        dispatch(requestStoprequest(tripId, stopId, requestType))

        let stopRequest = JSON.stringify({
            trip_id: tripId,
            stop_id: stopId,
            request_type: requestType
        })

        return fetch(config.API_URL + API_ENDPOINT, {
            method: 'POST',
            headers: {'Content-Type': 'application/json'},
            body: stopRequest
        })
            .then(response =>
            {
                if (response.ok)
                {
                    dispatch(receiveConfirm())

                    return response.json()
                }
                else
                {
                    dispatch(requestError())
                }
            })
            .catch(error => dispatch(requestError(error)))
    }
}
