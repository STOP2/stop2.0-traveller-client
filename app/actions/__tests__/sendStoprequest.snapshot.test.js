import mockStore from 'redux-mock-store'

import { requestError, receiveConfirm, requestStoprequest, setStopRequest_requestId } from '../sendStoprequest'

const store = mockStore()

beforeEach(() =>
{
    store.clearActions()
})

describe('sendStoprequest actions', () =>
{
    it('should handle REQUEST_ERROR action', () =>
    {
        expect(requestError()).toMatchSnapshot()
    })
})
