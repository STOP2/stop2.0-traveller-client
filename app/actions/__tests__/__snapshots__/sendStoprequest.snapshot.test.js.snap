exports[`sendStoprequest actions should handle REQUEST_ERROR action 1`] = `
Object {
  "error": true,
  "type": "REQUEST_ERROR",
}
`;
